x-pack/platform/plugins/shared/ml/public/application/model_management/trained_models_service.test.ts

```diff
@@ -0,0 +1,335 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import { BehaviorSubject, throwError, of } from 'rxjs';
+import type { Observable } from 'rxjs';
+import type { SavedObjectsApiService } from '../services/ml_api_service/saved_objects';
+import type {
+  StartAllocationParams,
+  TrainedModelsApiService,
+} from '../services/ml_api_service/trained_models';
+import { TrainedModelsService } from './trained_models_service';
+import type { TrainedModelUIItem } from '../../../common/types/trained_models';
+import { MODEL_STATE } from '@kbn/ml-trained-models-utils';
+import { i18n } from '@kbn/i18n';
+import type { MlTrainedModelConfig } from '@elastic/elasticsearch/lib/api/types';
+
+// Helper that resolves on the next microtask tick
+const flushPromises = () => new Promise<void>((resolve) => setImmediate(resolve));
+
+describe('TrainedModelsService', () => {
+  let mockTrainedModelsApiService: jest.Mocked<TrainedModelsApiService>;
+  let mockSavedObjectsApiService: jest.Mocked<SavedObjectsApiService>;
+  let trainedModelsService: TrainedModelsService;
+  let scheduledDeploymentsSubject: BehaviorSubject<StartAllocationParams[]>;
+  let mockSetScheduledDeployments: jest.Mock<any, any>;
+
+  const mockDisplayErrorToast = jest.fn();
+  const mockDisplaySuccessToast = jest.fn();
+
+  beforeEach(() => {
```

Comment:
Your unit tests shouldn’t wait for actual time to pass, as it increases test runtime. Instead, we should use fake timers.

```suggestion
  beforeEach(() => {
    jest.useFakeTimers();
```

Reply:
Done in https://github.com/elastic/kibana/pull/205699/commits/d6029a57c42b4582ac19c234cba1184b19980737